source/agora/common/BlockStorage.d

```diff
@@ -761,6 +788,80 @@ public class BlockStorage : IBlockStorage
         {
         }
     }
+
+    /*******************************************************************************
+
+        Read the bytes to calculate the checksum
+
+        Params:
+            data = file data bytes
+
+        Returns:
+            calculate the checksum
+
+    *******************************************************************************/
+
+    private ubyte[] checksum (const ubyte[] data) @safe nothrow
+    {
+        auto crc32 = new CRC32Digest();
+        crc32.put(data);
+        return crc32.finish();
+    }
+
+    /*******************************************************************************
+
+        Read the MemFile bytes and calculate them as checksums and
+        compare them with the expected checksum.
+
+        Returns:
+            Returns true if correct, otherwise returns false
+
+    *******************************************************************************/
+
+    private void validateChecksum () @trusted nothrow
+    {
+        try
+        {
+            const ubyte [] expected = cast(ubyte[])this.file[0 .. ChecksumSize];
+            const ubyte [] fileData = cast(ubyte[])this.file[ChecksumSize .. MapSize];
+            const actual = this.checksum(fileData);
+            if (expected != actual)
+            {
+                stderr.writefln("[ERROR] %s Block file is corrupted and will be terminated." , this.getFileName(this.file_index));
+                abort();
+            }
+        }
+        catch (Exception ex)
+        {
+            this.writeLog("BlockStorage.compareChecksum: ", ex);
+        }
+    }
+
+    /*******************************************************************************
+
+        Read the file data, calculate checksum,
+        and write checksum to file at start point
+
+        Returns:
+            Returns true if success, otherwise returns false.
+
+    *******************************************************************************/
+
+    private bool writeChecksum () @trusted nothrow
```

Comment:
Why is the return value of this function ignored?

Reply:
add assert(writeChecksum());